lib/jwt/verify.rb

```diff
@@ -53,9 +53,14 @@ def verify_iss
 
       iss = @payload['iss']
 
-      return if Array(options_iss).map(&:to_s).include?(iss.to_s)
+      options_iss = Array(options_iss).map { |item| item.is_a?(Symbol) ? item.to_s : item }
 
-      raise(JWT::InvalidIssuerError, "Invalid issuer. Expected #{options_iss}, received #{iss || '<none>'}")
+      case iss
+      when *options_iss
+        return
+      else
+        raise(JWT::InvalidIssuerError, "Invalid issuer. Expected #{options_iss}, received #{iss || '<none>'}")
```

Comment:
How will this error message look like when there are lambdas and methods in the mix?

Reply:
It will be something like `"Invalid issuer. Expected #Proc<...>, received foo_bar"`. Not the nicest case but there is no nice way to print out a proc. I also did not want to make extra cases.

If one wants a nicer output, they can define `to_s` on their proc:

```rb
matcher = ->(issuer) { issuer.start_with?('foo') && issuer != 'foo_bar' }
def matcher.to_s
  'A nicer description'
end
```